src/Exceptions/BulkRequestException.php

```diff
@@ -13,16 +13,47 @@ final class BulkRequestException extends ErrorException
 
     public function __construct(array $response)
     {
+        $this->response = $response;
+
         parent::__construct(
-            'One or more bulk operations did not complete successfully. ' .
-            'Catch the exception and use the BulkRequestException::getResponse() method to get more details.'
+            ($this->getFirstErrorFromResponse() ?? 'One or more bulk operations did not complete successfully.') .
+            ' Catch the exception and use the BulkRequestException::getResponse() method to get more details.'
         );
+    }
 
-        $this->response = $response;
+    public function context(): array
+    {
+        return [
+            'response' => $this->getResponse(),
+        ];
     }
 
     public function getResponse(): array
     {
         return $this->response;
     }
+
+    public function getResponseItems(): array
+    {
+        return $this->response['items'] ?? [];
+    }
+
+    private function getFirstErrorFromResponse(): ?string
```

Comment:
If it would be the `makeErrorFromResponse` method, we could do something like:

```php
$items = $this->response['items'];
$count = count($items);

$result = sprintf('%s did not complete successfully.', $count > 0 ? $count . ' bulk operation(s)' : 'One or more');
$firstError = $items[0]['error'] ?? null;

if (isset($firstError) && isset($firstError['type']) && isset($firstError['reason'])) {
    $result .= sprintf('%s: %s. Reason: %s.', $count > 1 ? 'First error' : 'Error', $firstError['type'], $firstError['reason']);
}

return sprintf('%s. Catch the exception and use the %s::getResponse() method to get more details.', $reason, self::class); 
```